src/com/itacademy/eshop/shop/Eshop.java

```diff
@@ -35,14 +35,15 @@ public void addProduct(Product shirt) {
         products.add(shirt);
     }
 
-    public void removeProductByName(String laptop) {
+    public Product removeProductByName(String laptop) {
```

Comment:
I think your method should remain void and you dont need to return anything? also parameter name `laptop` is too specific for a parameter which could have a different value than laptop. I suggest have it more like `productName`